indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -219,4 +269,75 @@ protected boolean isUseMaxMemoryEstimates()
         Tasks.DEFAULT_USE_MAX_MEMORY_ESTIMATES
     );
   }
+
+  protected ServiceMetricEvent.Builder getMetricBuilder()
+  {
+    return metricBuilder;
+  }
+
+  public IngestionMode getIngestionMode()
+  {
+    return ingestionMode;
+  }
+
+  protected static IngestionMode computeIngestionMode(@Nullable CompactionIOConfig ioConfig)
```

Comment:
Suggest distinguishing these two `computeIngestionMode` methods with different names since they both have nullable parameters

Reply:
Good call. Done in next commit.